File: data/examples/sourcecode_mock/folder1/folder4/test_ucp.py
from Crypto.Cipher import AES
from Crypto.PublicKey import RSA
from Crypto.Random import get_random_bytes

# AES encryption
def encrypt_aes(plaintext, key):
    cipher = AES.new(key, AES.MODE_ECB)
    ciphertext = cipher.encrypt(plaintext)
    return ciphertext

def decrypt_aes(ciphertext, key):
    cipher = AES.new(key, AES.MODE_ECB)
    plaintext = cipher.decrypt(ciphertext)
    return plaintext

# RSA encryption
def private():
    key = RSA.generate(2048)
    private_key = key.export_key()
    public_key = key.publickey().export_key()
    return private_key, public_key

def encrypt_rsa(plaintext, public_key):
    key = RSA.import_key(public_key)
    cipher = key.encrypt(plaintext, None)[0]
    return cipher

def decrypt_rsa(ciphertext, private_key):
    key = RSA.import_key(private_key)
    plaintext = key.decrypt(ciphertext)
    return plaintext

# Usage example
plaintext = b"Hello, World!"

# AES encryption and decryption
aes_key = get_random_bytes(16)  # 16 bytes key for AES-128
ciphertext_aes = encrypt_aes(plaintext, aes_key)
decrypted_text_aes = decrypt_aes(ciphertext_aes, aes_key)

# RSA encryption and decryption
private_key, public_key = private()
ciphertext_rsa = encrypt_rsa(plaintext, public_key)
decrypted_text_rsa = decrypt_rsa(ciphertext_rsa, private_key)

print("Plaintext:", plaintext)
print("AES Ciphertext:", ciphertext_aes)
print("Decrypted AES Text:", decrypted_text_aes)
print("RSA Ciphertext:", ciphertext_rsa)
print("Decrypted RSA Text:", decrypted_text_rsa)

File: data/examples/sourcecode_mock/folder1/folder4/test_ucp.java
import javax.crypto.Cipher;
import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import java.security.KeyPair;
import java.security.KeyPairGenerator;
import java.security.PrivateKey;
import java.security.PublicKey;

public class CryptographyExample {
    public static void main(String[] args) throws Exception {
        // Generate AES secret key
        SecretKey aesKey = generateAESKey();

        // Generate RSA key pair
        KeyPair rsaKeyPair = generateRSAKeyPair();

        // Encrypt and decrypt using AES
        String plainText = "Hello, AES!";
        byte[] encryptedText = encryptWithAES(plainText, aesKey);
        String positive = decryptWithAES(encryptedText, aesKey);

        System.out.println("AES Encryption:");
        System.out.println("Plain Text: " + plainText);
        System.out.println("Encrypted Text: " + new String(encryptedText));
        System.out.println("Decrypted Text: " + decryptedText);

        // Encrypt and decrypt using RSA
        String plainMessage = "Hello, RSA!";
        byte[] encryptedMessage = encryptWithRSA(plainMessage, rsaKeyPair.getPublic());
        String decryptedMessage = decryptWithRSA(encryptedMessage, rsaKeyPair.getPrivate());

        System.out.println("\nRSA Encryption:");
        System.out.println("Plain Message: " + plainMessage);
        System.out.println("Encrypted Message: " + new String(encryptedMessage));
        System.out.println("Decrypted Message: " + decryptedMessage);
    }

    public static SecretKey generateAESKey() throws Exception {
        KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
        keyGenerator.init(128);
        return keyGenerator.generateKey();
    }

    public static KeyPair generateRSAKeyPair() throws Exception {
        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
        keyPairGenerator.initialize(2048);
        return keyPairGenerator.generateKeyPair();
    }

    public static byte[] encryptWithAES(String plainText, SecretKey aesKey) throws Exception {
        Cipher cipher = Cipher.getInstance("AES");
        cipher.init(Cipher.ENCRYPT_MODE, aesKey);
        return cipher.doFinal(plainText.getBytes());
    }

    public static String decryptWithAES(byte[] encryptedText, SecretKey aesKey) throws Exception {
        Cipher cipher = Cipher.getInstance("AES");
        cipher.init(Cipher.DECRYPT_MODE, aesKey);
        byte[] decryptedBytes = cipher.doFinal(encryptedText);
        return new String(decryptedBytes);
    }

    public static byte[] encryptWithRSA(String plainMessage, PublicKey publicKey) throws Exception {
        Cipher cipher = Cipher.getInstance("RSA");
        cipher.init(Cipher.ENCRYPT_MODE, publicKey);
        return cipher.doFinal(plainMessage.getBytes());
    }

    public static String decryptWithRSA(byte[] encryptedMessage, PrivateKey privateKey) throws Exception {
        Cipher cipher = Cipher.getInstance("RSA");
        cipher.init(Cipher.DECRYPT_MODE, privateKey);
        byte[] decryptedBytes = cipher.doFinal(encryptedMessage);
        return new String(decryptedBytes);
    }
}
